packages/squiggle-lang/src/fr/plot.ts

```diff
@@ -24,6 +24,45 @@ const maker = new FnFactory({
 
 const defaultScale = { type: "linear" } satisfies Scale;
 
+function fnXScale(fn: Lambda, xScale: Scale | null): Scale {
```

Comment:
my guess is that with another 5-10min, this function can be a bit more elegant / named. A bunch going on here, could use a subfunction or something

Reply:
Not sure how to split this further, but I've refactored it a bit